parlay/items/base.py

```
@@ -61,12 +61,23 @@ class BaseItem(object):
     The Base Item that all other Items should inherit from
     """
 
-    def __init__(self, item_id, name, adapter=None):
+    def __init__(self, item_id, name, adapter=None, parents=None):
         self.item_id = item_id
         self.item_name = name
         """:type Adapter"""  # use the default pyadapter if no specific adapter was chosen
         self._adapter = adapter if adapter is not None else Broker.get_instance().pyadapter
         self.children = []  # child items
+        self.parents = [] if not parents else parents
+        self.parents = self.parents if isinstance(self.parents, Iterable) else [self.parents]
+
+        if self.parents:
+            try:
+                for parent in self.parents:
+                    parent.add_child(self)
+            except Exception as e:
```

We should re-raise the Exception here instead of eating it. That way in debug mode on init people will know if they messed up

---

Fixed